dotnet/src/SemanticKernel.Abstractions/Orchestration/ContextVariables.cs

```diff
@@ -21,15 +14,19 @@ namespace Microsoft.SemanticKernel.Orchestration;
 /// </summary>
 [DebuggerDisplay("{DebuggerDisplay,nq}")]
 [DebuggerTypeProxy(typeof(ContextVariables.TypeProxy))]
-public sealed class ContextVariables : IDictionary<string, string>
+public sealed class ContextVariables : Dictionary<string, string>
 {
     /// <summary>
     /// Constructor for context variables.
     /// </summary>
     /// <param name="value">Optional value for the main variable of the context including trust information.</param>
     public ContextVariables(string? value = null)
+        : base(StringComparer.OrdinalIgnoreCase)
     {
-        this._variables[MainKey] = value ?? string.Empty;
+        if (value != null)
+        {
+            this[MainKey] = value;
+        }
```

Comment:
Call `Set(MainKey, value)` to ensure consistency.

```suggestion
        this.Set(MainKey, value);
```